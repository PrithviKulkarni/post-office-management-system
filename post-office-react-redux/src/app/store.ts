import { configureStore, ThunkAction, Action } from '@reduxjs/toolkit';
import letterReducer from '../features/LetterSlice';
import parcelReducer from '../features/ParcelSlice';
import depotReducer from '../features/DepotSlice';
import driverReducer from '../features/DriverSlice' 
export const store = configureStore({
  reducer: {
    letter: letterReducer,
    parcel: parcelReducer,
    depot: depotReducer,
    driver: driverReducer
  },
});

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;
