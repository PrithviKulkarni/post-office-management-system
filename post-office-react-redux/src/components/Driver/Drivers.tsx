import { Link } from "react-router-dom";
import { Accordion } from "react-bootstrap";
import { getAllDrivers } from "../../features/DriverSlice";
import { useAppSelector } from "../../app/hooks";

const Drivers = () => {
    const drivers = useAppSelector(getAllDrivers);
    console.log({ drivers });


    /**
     * *
     * Loads an array of Players Games and displays them a table.
     *
     *  @returns array of Players Games. */

    return (

        <div className="drivers" data-testid="drivers-page">
            <Accordion defaultActiveKey="0">
                {drivers.map(driver => (
                    <Accordion.Item eventKey={driver.id} className="accordian-item" data-testid="accordian-item">
                        <Accordion.Header><h3>{driver.first_name} &nbsp; {driver.last_name}</h3></Accordion.Header>
                        <Accordion.Body className="bg-warning">                                    <div>
                            <Link style={{ color: "black" }} to={`/updateDriver/${driver.id}`}>
                                Update Driver
                            </Link> &nbsp;
                            <Link style={{ color: "black" }} to={`/deleteDriver/${driver.id}`}>
                                Delete Driver
                            </Link>
                        </div>
                        </Accordion.Body>
                    </Accordion.Item>
                ))}
            </Accordion>
        </div>
    );
};

export default Drivers;
