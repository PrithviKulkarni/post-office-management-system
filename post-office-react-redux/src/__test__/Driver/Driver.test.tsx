import { render, cleanup } from '@testing-library/react';
import { screen } from '@testing-library/react';
import { Provider } from "react-redux";
import { store } from '../../app/store';
import Drivers from '../../components/Driver/Drivers';

afterEach(cleanup);

// DRIVERS COMPONENT

it('Renders Drivers Page Correctly', () => {
     const { asFragment } = render(
          <Provider store={store}>
               <Drivers />
          </Provider>
     );
     expect(asFragment()).toMatchSnapshot();
});

it('Should Have Div ID drivers-page', () => {
     render(
          <Provider store={store}>
               <Drivers />
          </Provider>);
     expect(screen.getByTestId("drivers-page")).toBeInTheDocument();
})