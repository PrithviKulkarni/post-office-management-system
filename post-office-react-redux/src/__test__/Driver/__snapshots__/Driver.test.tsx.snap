// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Drivers Page Correctly 1`] = `
<DocumentFragment>
  <div
    class="drivers"
    data-testid="drivers-page"
  >
    <div
      class="accordion"
    />
  </div>
</DocumentFragment>
`;
